src/Editor.js

```diff
@@ -514,8 +515,36 @@ const Editor = () => {
                   true
                 );
               }}
+              onDoubleClick={() => setEditableListItemId(id)}
+              onBlur={() => setEditableListItemId("")}
             >
-              {name}
+              {editableListItemId === id ? (
+                <input
+                  type="text"
+                  autoFocus
+                  onChange={event => {
+                    event.persist();
+                    const updatedName = event.currentTarget.value;
+                    setState(current => ({
+                      ...current,
+                      doc: {
+                        ...current.doc,
+                        layers: current.doc.layers.map(layer => {
+                          if (layer.id === id) {
+                            const updatedLayer = layer;
+                            updatedLayer.name = updatedName;
+                            return updatedLayer;
+                          }
+                          return layer;
+                        })
+                      }
+                    }));
+                  }}
+                  value={name}
+                />
+              ) : (
+                <>{name}</>
```

Comment:
We should be able to just return a string here because it's a legit React "node" (null, a boolean, a number, a string, a React element, or an array of any of those types recursively):

```suggestion
                name
```

We only need to use a fragment (which is just a special React element) when we want to return multiple nodes, but don't want to put them into an array.